### PR TITLE
Don't trash errno in bam_tag2cigar.

### PR DESCRIPTION
**Pull request by jkbonfield**
bam_aux_get sets errno=ENOENT when we look for a tag and don't find it.  Personally I think this is a misfeature, but it's part of the
documented API so we can't change that. Instead we undo the damage as not finding the tag is not an error.

The consequence of samtools/samtools#1296 is that now all almost errors found while decoding a BAM file get converted to "No such file or directory" prior to this commit.

### CODE REVIEW

File: sam.c [hunk 1]
```diff
@@ -522,7 +522,10 @@ static int bam_tag2cigar(bam1_t *b, int recal_bin, int give_warning) // return 0
     cigar0 = bam_get_cigar(b);
     if (bam_cigar_op(cigar0[0]) != BAM_CSOFT_CLIP || bam_cigar_oplen(cigar0[0]) != c->l_qseq) return 0;
     fake_bytes = c->n_cigar * 4;
-    if ((CG = bam_aux_get(b, "CG")) == 0) return 0; // no CG tag
+    int saved_errno = errno;
+    CG = bam_aux_get(b, "CG");
+    errno = saved_errno;
```
Comment (daviesrob) [inline sam.c:527]: This should really test for and propagate the `errno == EINVAL` case when `GC == NULL`, instead of glossing over it.

Comment (jkbonfield) [reply sam.c:527]: Ok, something like

```
if (errno == ENOENT) errno = saved_errno;
```

I'll amend when I'm finished with my current other PR, or make the trivial change during merge if you wish.

Comment (daviesrob) [reply sam.c:527]:
```
if (!CG) {
    if (errno == EINVAL) return -1;  // Bad aux data
    errno = saved_errno;
    return 0; // no CG tag
}
```

**Comment by daviesrob** [comment]
Rebased with fixed return value in the bad aux data error case.

**Pull request merged**